### PR TITLE
docs: add doc for GetIsReal

### PR DESCRIPTION
**Pull request by peter7891**
Adding doc to `Object::GetIsReal()` based on an explanation from @jaekwon

### CODE REVIEW

File: gnovm/pkg/gnolang/ownership.go
Comment (moul) [inline gnovm/pkg/gnolang/ownership.go:101]:
```suggestion
	// GetIsReal determines the reality of an Object: unpersisted state is fake, while persisted state is real.
	// This concept reflects a metaphysical understanding, where proof and persistence define an object's reality.
```

**Review by moul — APPROVED** [review]

**Comment by peter7891** [comment]
@moul should i merge?

**Comment by moul** [comment]
Yep, I reviewed, small change; no problem.

I like letting it open so anyone in the team has choice to make a second review and merge.

**Review by harry-hov — APPROVED** [review]
I prefer one line comment on declaration and detailed comment on definition.

Somthing like:
```go

type Object interface {
	...
	// SetOwner sets the owner of the Object.
	SetOwner(Object) 
	// GetIsOwned checks if the Object is owned.
	GetIsOwned() bool
	// GetIsReal determines the reality of an Object.
	GetIsReal() bool
	// GetModTime returns the modification time of the Object.
	GetModTime() uint64
	...
}

// GetIsReal determines the reality of an Object. During a transaction, the object 
// is fake, but becomes real upon successful completion, making it persisted and 
// verifiable. This concept reflects a metaphysical understanding, where proof and 
// persistence define an object's reality.
// NOTE: does not return true for new reals.
func (oi *ObjectInfo) GetIsReal() bool {
	return !oi.ID.IsZero()
}
```

**Comment by thehowl** [comment]
> I prefer one line comment on declaration and detailed comment on definition. (I have no problems with this PR though, it can be merged)

I think this works better if you reference the full documentation in the shorter documentation.

**Comment by thehowl** [comment]
Sorry, accidentally closed

File: gnovm/pkg/gnolang/ownership.go
Comment (thehowl) [inline gnovm/pkg/gnolang/ownership.go:102]: I think we need to decide on whether to have long docs on the interface or on the method (of ObjectInfo). The other one should refer to the one with the long doc. I'm fine either way, but I slightly lean towards having the long doc here instead of ObjectInfo, as we mostly use the Object interface instead of ObjectInfo.

In either case, I think the longer doc, whichever it is, that should additionally have the note currently in the func doc of `ObjectInfo.GetIsReal`:

```
// NOTE: does not return true for new reals.
```

**Comment by moul** [comment]
Merging now, because the content is good.

Feel free to make an overall doc improvement that will make the whole thing more consistent and usable 👍

**Pull request merged**
